src/init.cpp

```diff
@@ -275,6 +275,7 @@ void DynamicDataLoader::initialize()
     add( "field_type", &field_types::load );
     add( "field_type_migration", &field_type_migrations::load );
     add( "weather_type", &weather_types::load );
+    add( "weather_generator", &weather_generator::load_weather_generator );
```

Comment:
Can you please also add a finalize which calls finalize on the generic_factory? Unresolved deferred JSON will not be handled otherwise.

Reply:
You still need to add the call to it to init.cpp.